zypper.py

```diff
@@ -0,0 +1,274 @@
+"""
+
+import argparse
+import subprocess
+import os
+import sys
+
+from collections.abc import Sequence
+from prometheus_client import CollectorRegistry, Gauge, Info, generate_latest
+
+REGISTRY = CollectorRegistry()
+NAMESPACE = "zypper"
+
+
+def __print_pending_data(data, fields, info, filters=None):
+    filters = filters or {}
+
+    if len(data) == 0:
+        field_str = ",".join([f'{name}=""' for _, name in fields])
+        info.info({field_str: '0'})
+    else:
+        for package in data:
+            check = all(package.get(k) == v for k, v in filters.items())
+            if check:
+                field_str = ",".join([f'{name}="{package[field]}"' for field, name in fields])
+                info.info({field_str: '1'})
+
+
+def print_pending_updates(data, all_info, filters=None):
+    if all_info:
+        fields = [("Repository", "repository"),
+                  ("Name", "package-name"),
+                  ("Available Version",
+                   "available-version")]
+    else:
+        fields = [("Repository", "repository"),
+                  ("Name", "package-name")]
+    prefix = "zypper_update_pending"
+    description = (
+        "zypper package update available from repository. "
+        "(0 = not available, 1 = available)"
+    )
+    info = Info(prefix, description)
+
+    __print_pending_data(data, fields, info, filters)
+
+
+def print_pending_patches(data, all_info, filters=None):
+    if all_info:
+        fields = [("Repository", "repository"),
+                  ("Name", "patch-name"),
+                  ("Category", "category"),
+                  ("Severity", "severity"),
+                  ("Interactive", "interactive"),
+                  ("Status", "status")]
+    else:
+        fields = [("Repository", "repository"),
+                  ("Name", "patch-name"),
+                  ("Interactive", "interactive"),
+                  ("Status", "status")]
+    prefix = "zypper_patch_pending"
+    description = "zypper patch available from repository. (0 = not available , 1 = available)"
+    info = Info(prefix, description)
+
+    __print_pending_data(data, fields, info, filters)
+
+
+def print_orphaned_packages(data, filters=None):
+    fields = [("Name", "package"),
+              ("Version", "installed-version")]
+    prefix = "zypper_package_orphan"
+    description = "zypper packages with no update source (orphaned)"
+    info = Info(prefix, description)
+
+    __print_pending_data(data, fields, info, filters)
+
+
+def print_data_sum(data, prefix, description, filters=None):
+    gauge = Gauge(prefix,
+                  description,
+                  namespace=NAMESPACE,
+                  registry=REGISTRY)
+    filters = filters or {}
+    if len(data) == 0:
+        gauge.set(0)
+    else:
+        for package in data:
+            check = all(package.get(k) == v for k, v in filters.items())
+            if check:
+                gauge.inc()
+
+
+def print_reboot_required():
```

Comment:
This overlaps with an existing textfile collector in this repo, `needrestart_info.py`, and essentially does not belong in this collector.